Automattic/jetpack
# WordPress.com Features: Calypso Locale Sync from Calypso to wp-admin

Fixes https://github.com/Automattic/dotcom-forge/issues/6812
Related https://github.com/Automattic/jetpack/pull/37316 https://github.com/Automattic/dotcom-forge/issues/7086

## Proposed changes:

* Use hook `admin_init` to check whether the wp-admin locale needs to be synced with Calypso locale
* Duplicate code for locale sync from https://github.com/Automattic/jetpack/pull/19525

>[!Note]
> The masterbar module contains [the logic](https://github.com/Automattic/jetpack/blob/839954e7a11ac72b670010dbe4eb1e0f894e6f66/projects/plugins/jetpack/modules/masterbar/masterbar/class-masterbar.php#L140) for syncing the Calypso locale in Atomic Default. It's loaded [here](https://github.com/Automattic/jetpack/blob/839954e7a11ac72b670010dbe4eb1e0f894e6f66/projects/plugins/jetpack/modules/masterbar.php#L27), only when the navigation redesign flag is disabled.
> There is a project in progress that is moving Jetpack Masterbar module to mu-wpcom. See https://github.com/Automattic/jetpack/pull/37277 and read more in pfwV0U-3U-p2 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to https://wordpress.com/me/account to change Calypso locale
* Visit any admin page on your atomic site with the nav redesign option flag enabled
* Verify the Calypso locale appears selected in the dropdown in `/wp-admin/profile.php`
* Verify the locale is applied on the whole UI on the first load
